src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5358,9 +5358,11 @@ void Parker::unpark() {
 // Platform Monitor implementation
 
 // Must already be locked
-int PlatformMonitor::wait(jlong millis) {
-  assert(millis >= 0, "negative timeout");
+int PlatformMonitor::wait(uint64_t millis) {
   int ret = OS_TIMEOUT;
+  if (millis > UINT_MAX) {
```

Comment:
Can you add a one line comment saying why this is?  is it a platform limitation? or just would be mostly infinite if bigger than UINT_MAX?

Reply:
The platform API actually takes a DWORD aka uint32_t. I will add a comment.